build.gradle.kts

```
@@ -2,10 +2,48 @@ plugins {
     //trick: for the same plugin versions in all sub-modules
     id("com.android.application").version("8.0.2").apply(false)
     id("com.android.library").version("8.0.2").apply(false)
+    id("io.gitlab.arturbosch.detekt").version("1.23.0")
     kotlin("android").version("1.8.21").apply(false)
     kotlin("multiplatform").version("1.8.21").apply(false)
 }
 
+detekt {
+    toolVersion = "1.23.0"
+    config.setFrom("detekt.yml")
+
+    source.setFrom(
+        "android/src/main/java",
+        "shared/src/androidMain/kotlin",
+        "shared/src/androidUnitTest/kotlin",
+        "shared/src/commonMain/kotlin",
+        "shared/src/commonTest/kotlin",
```

IMO, this should be simply this to ensure checking issues on test area too

```suggestion
        "android/src",
        "shared/src"
```

---

@luongvo After trying, The detekt cannot run test for `"shared/src/iosTest/kotlin"` directory and it causes the detekt `BUILD FAILED`. Do you have any idea how to fix it? 🙏🏻